dtcli/api.py

```diff
@@ -0,0 +1,108 @@
+        r.raise_for_status()
+        return r.json()["items"]
+
+    def acquire_environment_configuration(self, fqdn: str):
+        r = self.requests.get(self.url_base + f"/api/v2/extensions/{fqdn}/environmentConfiguration", headers=self.headers)
+
+        if r.status_code == 404:
+            return
+
+        r.raise_for_status()
+        return r.json()
+
+    def acquire_extensions(self):
+        r = self.requests.get(self.url_base + f"/api/v2/extensions", headers=self.headers)
+        r.raise_for_status()
+        return r.json()["extensions"]
+
+    def acquire_extension_versions(self, fqdn: str):
+        r = self.requests.get(self.url_base + f"/api/v2/extensions/{fqdn}", headers=self.headers)
+
+        r.raise_for_status()
+        return r.json()["extensions"]
+
+    def delete_monitoring_configuration(self, fqdn: str, configuration_id: str):
+        r = self.requests.delete(self.url_base + f"/api/v2/extensions/{fqdn}/monitoringConfigurations/{configuration_id}", headers=self.headers)
+        try:
+            r.raise_for_status()
+        except:
+            err = ""
+            try:
+                err = r.json()
+            except:
+                pass
+
+            print(err)
+            raise
+
+    def delete_environment_configuration(self, fqdn: str):
+        r = self.requests.delete(self.url_base + f"/api/v2/extensions/{fqdn}/environmentConfiguration", headers=self.headers)
+        err = r.json()
+        try:
+            r.raise_for_status()
+        except:
+            print(err)
+            if r.code != 404:
+                raise
+
+    def delete_extension(self, fqdn: str, version: str):
+        r = self.requests.delete(self.url_base + f"/api/v2/extensions/{fqdn}/{version}", headers=self.headers)
+        err = r.json()
+        try:
+            r.raise_for_status()
+        except:
+            print(err)
+            if r.code != 404:
+                raise
+
+    def get_schema_target_version(self, target_version: str):
+        """Get version number from tenant. If version doesn't exist return list of available versions."""
+        versions = self.requests.get(self.url_base + "/api/v2/extensions/schemas", headers=self.headers).json().get("versions", [])
+
+        if target_version == "latest":
+            return versions[-1]
+
+        matches = [v for v in versions if v.startswith(target_version)]
+        if matches:
+            return matches[0]
+
+        raise SystemExit(f"Target version {target_version} does not exist. \nAvailable versions: {versions}")
+
+    def download_schemas(self, target_version: str, download_dir: str):
+        """Downloads schemas from choosen version"""
+
+        version = self.get_schema_target_version(target_version)
+
+        if not os.path.exists(download_dir):
+            os.makedirs(download_dir)
+
+        files = self.requests.get(self.url_base + f"/api/v2/extensions/schemas/{version}", headers=self.headers).json().get("files", [])
```

Comment:
You don't need this. If you query for `application/octet-stream` a ZIP archive with all the files for the given version will be returned.